### PR TITLE
Fix array hashCode calculation in ResyncReplicationRequest

### PR DESCRIPTION
**Pull request by dk2k**
…de(). Added unit test
Previous reincarnation of this PR https://github.com/opensearch-project/OpenSearch/pull/15383

### CODE REVIEW

**Review by dbwiddis — APPROVED** [review]
Code changes LGTM (see also review on previous PR).

Will let CI finish and another maintainer review before merging.

**Review by dbwiddis — APPROVED** [review]

**Review by owaiskazi19 — APPROVED** [review]

**Pull request merged**
